Python/Product/PythonTools/PythonTools/Refactoring/ExtractMethodRequestView.cs

```diff
@@ -237,14 +238,38 @@ public ExtractMethodRequestView(IServiceProvider serviceProvider, ExtractedMetho
         /// </summary>
         void ExtractMethodRequestView_PropertyChanged(object sender, PropertyChangedEventArgs e) {
             if (e.PropertyName != "IsValid") {
-                IsValid = (Name != null && _validNameRegex.IsMatch(Name)) &&
-                    TargetScope != null;
+                IsValid = (TargetScope != null) && IsValidPythonIdentifier(Name, _previewer.PythonVersion);
             }
             if (e.PropertyName != "PreviewText") {
                 UpdatePreview();
             }
         }
 
+        internal static bool IsValidPythonIdentifier(string identifier, PythonLanguageVersion pythonVersion) {
+            if (String.IsNullOrEmpty(identifier) || PythonKeywords.IsKeyword(identifier, pythonVersion)) {
+                return false;
+            }
+
+            //Python2 identifiers are only certain ASCII characters
+            if (pythonVersion < PythonLanguageVersion.V30) {
+                var Python2IdentifierRegex = new Regex("^[a-zA-Z_][a-zA-Z0-9_]*$");
```

Comment:
I would keep the regex as a private static field (similar to what it was before but private) so it doesn't get new'ed every time